packages/@aws-cdk/aws-ecs-patterns/lib/ecs/queue-processing-ecs-service.ts

```diff
@@ -111,6 +111,12 @@ export class QueueProcessingEc2Service extends QueueProcessingServiceBase {
       enableECSManagedTags: props.enableECSManagedTags,
       deploymentController: props.deploymentController,
     });
+
+    if (!this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT)) {
+      const cfnService = this.service.node.children.find(c => c instanceof CfnService) as CfnService;
+      cfnService.desiredCount = this.desiredCount;
+    }
+
```

Comment:
Can't we just set it via the props? Manually escape hatching in this way is not common in constructors.
```ts
const desiredCount = this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT)
  ? this.desiredCount
  : undefined;

this.service = new Ec2Service(this, 'QueueProcessingService', {
  desiredCount,
  ...
});
```

Reply:
That's fair, we can set it as part of the construct, but we want to remove the property with the feature flag (would you prefer setting it undefined, in v2 or removing it altogether?)

Reply:
Setting it to undefined is essentially the same as not passing it. IE the property doesn't get serialized into the template. Its one more place to delete code in V2 once the flag is removed, but I prefer that to reaching in like this.

Reply:
Addressed